core/embed/sec/tropic/tropic_transport.c

```diff
@@ -0,0 +1,60 @@
+/*
+ * This file is part of the Trezor project, https://trezor.io/
+ *
+ * Copyright (c) SatoshiLabs
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+
+#include <trezor_rtl.h>
+#include <sec/tropic_transport.h>
+#include <sec/secret.h>
+#include "ed25519-donna/ed25519.h"
+#include "memzero.h"
+
+#define PKEY_INDEX_BYTE PAIRING_KEY_SLOT_INDEX_0
+
+STATIC lt_handle_t lt_handle = {0};
+
+tropic_result tropic_init(void) {
+  uint8_t tropic_secret_tropic_pubkey[SECRET_TROPIC_KEY_LEN] = {0};
+  uint8_t tropic_secret_trezor_privkey[SECRET_TROPIC_KEY_LEN] = {0};
+
+  ensure((lt_init(&lt_handle) == LT_OK) * sectrue, "lt_init failed");
+  ensure(secret_tropic_get_tropic_pubkey(tropic_secret_tropic_pubkey), "secret_tropic_get_tropic_pubkey failed");
+  ensure(secret_tropic_get_trezor_privkey(tropic_secret_trezor_privkey), "secret_tropic_get_trezor_privkey failed");
+
+  tropic_result result = tropic_handshake(tropic_secret_trezor_privkey, tropic_secret_tropic_pubkey);
+
+  memzero(tropic_secret_trezor_privkey, sizeof(tropic_secret_trezor_privkey));
+
+  ensure((result == TROPIC_SUCCESS) * sectrue, "tropic_handshake failed");
+
+  return TROPIC_SUCCESS;
+}
+
+tropic_result tropic_handshake(const uint8_t *trezor_privkey, const uint8_t *tropic_pubkey) {
+  lt_ret_t ret = LT_FAIL;
+
+  uint8_t trezor_pubkey[SECRET_TROPIC_KEY_LEN] = {};
+  curve25519_scalarmult_basepoint(trezor_pubkey, trezor_privkey);
+
+  ret = lt_session_start(&lt_handle, tropic_pubkey, PKEY_INDEX_BYTE, trezor_privkey, trezor_pubkey);
+  if (ret != LT_OK) {
+    return TROPIC_ERR_SESSION_START;
+  }
+
+  return TROPIC_SUCCESS;
+}
```

Comment:
I would consider inlining this function.

Reply:
Done!

Reply:
Oh, it seems the term "to inline a function" is ambiguous. You meant to use the `inline` qualifier and I meant to replace the call with the contents of the function.

The reason I suggested this was that `tropic_handshake` makes two calls, and the rest of the function is error handling. It seems to me that inlining could make it easier to read.

Reply:
Yeah, I don't know how I understood the one that made least amount of sense. Anyway... https://github.com/trezor/trezor-firmware/pull/4172/commits/07425d69411a20883084721064158b4c5a70e8e5#diff-2179dea73b04181bd25219fedb2e591db4a1cbd46972c416eefd1f1adf882f45R43